File: backend/app/schemas/notification.py
"""Pydantic schemas for in-app notifications."""
import uuid
from datetime import datetime

from pydantic import BaseModel


class NotificationPublic(BaseModel):
    id: uuid.UUID
    notification_type: str
    title: str
    message: str
    is_read: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class PaginatedNotifications(BaseModel):
    items: list[NotificationPublic]
    total: int
    unread_count: int
    page: int
    page_size: int

File: frontend/src/types/notification.ts
export interface NotificationPublic {
  id: string;
  notification_type: string;
  title: string;
  message: string;
  is_read: boolean;
  created_at: string;
}

export interface PaginatedNotifications {
  items: NotificationPublic[];
  total: number;
  unread_count: number;
  page: number;
  page_size: number;
}
